python/tests/test_topology.py

```diff
@@ -48,6 +48,293 @@ def generate_segments(n, sequence_length=100, seed=None):
         segs.append(tests.Segment(left, right, j))
     return segs
 
+def kc_distance_tree(tree1, tree2, lambda_param = 0):
+    """
+    Returns the Kendall-Colijn distance between the specified
+    pair of trees. Lambda parameter determines weight of branch lengths vs. topology in calculating the distance.
+    """
+    samples = tree1.tree_sequence.samples()
+    if not np.array_equal(samples, tree2.tree_sequence.samples()):
+        raise ValueError("Trees must have the same samples")
+    k = samples.shape[0]
+    n = (k * (k - 1)) // 2
+    m = [np.ones(n + k), np.ones(n + k)]
+    M = [np.zeros(n + k), np.zeros(n + k)]
+    for tree_index, tree in enumerate([tree1, tree2]):
+        stack = [(tree.root, 0, tree.time(tree.root))]
+        while len(stack) > 0:
+            u, depth, time = stack.pop()
+            children = tree.children(u)
+            for v in children:
+                stack.append((v, depth + 1, tree.time(v)))
+            for c1, c2 in itertools.combinations(children, 2):
+                for u in tree.samples(c1):
+                    for v in tree.samples(c2):
+                        if u < v:
+                            a, b = u, v
+                        else:
+                            a, b = v, u
+                        pair_index = a * (a - 2 * k + 1) // -2 + b - a - 1
+                        assert m[tree_index][pair_index] == 1
+                        m[tree_index][pair_index] = depth
+                        M[tree_index][pair_index] = tree.time(tree.root) - time
+            if len(tree.children(u)) == 0:
+                M[tree_index][u + n] = tree.time(tree.parent(u))
+    return ((1 - lambda_param) * np.linalg.norm(m[0] - m[1])) + (lambda_param * np.linalg.norm(M[0] - M[1]))
+
+def kc_distance_ts(ts_1, ts_2, lambda_param = 0):
```

Comment:
Let's keep it to just trees for the PR @awohns, we can think about the TS case later.